include/Animal.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include "Location.h"
+#include <iostream>
+#include <vector>
+#include <map>
+using namespace std;
+#include <cstdlib> // for rand()
+#include <ctime>   // for time()using namespace std;
+class Animal
+{
+protected:
+	string _name;
+	Location _location;
+	bool _stopped = true;
+
+public:
+	Animal(const string&, const Location&);
+	virtual void printDetails() const = 0;
```

Comment:
no use of virtual destructor 
check this
https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors